_posts/2024-05-27-A-deep-dive-into-faster-semantic-sparse-retrieval-in-OS-2.12.md

```diff
@@ -0,0 +1,144 @@
+
+For medium to large datasets (millions to tens of millions of documents), neural sparse search is very efficient, with smaller client-side latency and significantly lower resource consumption compared to dense vector search (see [this blog](https://opensearch.org/blog/improving-document-retrieval-with-sparse-semantic-encoders/)). The doc-only mode delivers a better client-side search latency, faster than the time required for machine learning (ML) inference. Neural sparse search provides additional cost savings because you don't need to host a powerful ML service that supports a high throughput. 
+
+However, for extremely large datasets (billions of documents), the latency of neural sparse search also grows linearly with the number of documents. In this case, a dense vector search is a better choice for low latency, because the latency for the underlying Approximate Nearest Neighbor (ANN) algorithm used in dense vector search grows logarithmically with the number of documents. We're actively working on improvements to neural sparse search to address its scalability limitations for extremely large datasets.
+
+## Accelerating sparse search with Lucene upgrades
+
+In the past 20 years, Lucene has continuously optimized its core performance, from streamlining index file I/O to improving posting list merging algorithms. Building neural sparse search on top of Lucene allowed us to take advantage of these optimizations. The optimizations are ongoing, and OpenSearch 2.12 uses the latest advancements in Lucene 9.9. Compared to Lucene 9.7 used in OpenSearch 2.11, Lucene 9.9 offers a significant performance boost, making it the fastest release yet (for more information, see the [Lucene release notes](https://cwiki.apache.org/confluence/display/LUCENE/Release+Notes)). This improved performance extends to neural sparse search.
+
+Because of its use of the top-k hits disjunctive Boolean query, **neural sparse search performance improves with the new Lucene version**. To demonstrate this, we conducted a performance comparison between OpenSearch 2.11 and 2.12 on the MS MARCO dataset. We used OpenSearch [pretrained sparse encoding models](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sparse-encoding-models) on the same cluster setup mentioned previously, with each index having 3 primary shards and 1 replica shard. All indexes were force-merged before the search for consistency.
```

Comment:
```suggestion
Because of its use of the top-k hits disjunctive Boolean query, **neural sparse search performance improves with the new Lucene version**. To demonstrate this, we conducted a performance comparison between OpenSearch 2.11 and 2.12 on the MS MARCO dataset. We used OpenSearch [pretrained sparse encoding models](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sparse-encoding-models) on the same cluster setup mentioned previously, with each index having 3 primary shards and 1 replica shard. All indexes were force merged before the search in order to ensure consistency.
```